src/modm/math/algorithm/prescaler.hpp

```diff
@@ -29,50 +27,47 @@ namespace modm
  * @note For ranges the end is *inclusive*: [begin, end]!
  * @ingroup	modm_math_algorithm
  */
-template<std::unsigned_integral T>
+template<typename T>
 class
 GenericPrescaler
 {
 public:
 	struct Result
 	{
-		constexpr Result(T input_frequency,
-		                 T desired_frequency,
-		                 uint32_t index, uint32_t prescaler) :
+		constexpr Result(T input_frequency, T desired_frequency,
+						 uint32_t index, uint32_t prescaler) :
 			index{index}, prescaler{prescaler},
 			frequency{input_frequency / prescaler},
 			input_frequency{input_frequency},
 			desired_frequency{desired_frequency},
-			error{1.f - float(input_frequency / prescaler) / desired_frequency}
+			error{1.f - float(input_frequency) / float(prescaler * desired_frequency)}
 		{}
 		/// Zero-indexed prescaler result
-		const uint32_t index;
+		uint32_t index;
 		/// Prescaler value
-		const uint32_t prescaler;
+		uint32_t prescaler;
 		/// Generated frequency
-		const T frequency;
+		T frequency;
 		/// Input frequency
-		const T input_frequency;
+		T input_frequency;
 		/// Desired Frequency
-		const T desired_frequency;
+		T desired_frequency;
 		/// Relative Frequency Error
-		const float error;
+		float error;
 	};
 
 public:
-	/// From any iterable container.
-	/// @note container must have `begin()`, `end()` and `size()` function!
-	template<class Iterator>
+	/// From any forward range.
```

Comment:
The function also requires the forward range to be sized because it calls `std::ranges::size()`. This wouldn't work in cases where computing the size is O(n) like for a `std::forward_list` or with certain range adaptors. `std::ranges::distance()` would be the more generic function to compute the size even if it's O(n). Just a little nitpick, the function is probably fine as is.

Furthermore this function has undefined behavior if an empty range is passed in, but this case doesn't make sense. Not sure if we can do anything about it. When used in a compile-time context it simply wouldn't compile.